### PR TITLE
refactor(catalog): catalog initialisation refactor

### PR DESCRIPTION
**Pull request by praveen-influx**
- when no catalog is found, create a new one with instance id and persist it immediately
- enabled test-log in influxdb3_write

Closes: https://github.com/influxdata/influxdb/issues/25346

### CODE REVIEW

File: influxdb3_write/src/persister.rs [hunk 1]
```diff
@@ -120,25 +120,23 @@ impl Persister {
...
-    ) -> Result<(LastCacheProvider, Catalog)> {
-        match self.load_catalog().await? {
-            Some(c) => Ok((
-                LastCacheProvider::new_from_catalog(&c.catalog)?,
-                Catalog::from_inner(c.catalog),
-            )),
+    /// Try loading the catalog, if there is no catalog generate new
+    /// instance id and create a new catalog and persist it immediately
+    pub async fn load_or_create_catalog(&self) -> Result<Catalog> {
+        let catalog = match self.load_catalog().await? {
```
Comment (praveen-influx) [inline influxdb3_write/src/persister.rs:126]: @pauldix This is slightly different to what has been mentioned in the [issue](https://github.com/influxdata/influxdb/issues/25346), I'm reusing the `load_catalog` method which was already in use to determine whether to generate instance id or not. If this is not ideal, happy to rework this PR.

File: influxdb3_write/src/persister.rs [hunk 2]
```diff
@@ -120,25 +120,23 @@ impl Persister {
...
+            Some(c) => Catalog::from_inner(c.catalog),
             None => {
                 let uuid = Uuid::new_v4().to_string();
                 let instance_id = Arc::from(uuid.as_str());
-                Ok((LastCacheProvider::new(), Catalog::new(host_id, instance_id)))
+                info!("Created new instance id {:?}", instance_id);
```
Comment (hiltontj) [inline influxdb3_write/src/persister.rs:131]:
```suggestion
                info!(instance_id = ?instance_id, "catalog not found, creating new instance id");
```
Just encouraging use of `tracing`'s field syntax.

This is somewhat redundant with the `info!` emitted at the caller level, but I don't think it hurts to have.

File: influxdb3_write/src/write_buffer/mod.rs [hunk 3]
```diff
@@ -1150,7 +1131,7 @@ mod tests {
         )
         .await;
 
-        verify_catalog_count(1, write_buffer.persister.object_store()).await;
+        verify_catalog_count(2, write_buffer.persister.object_store()).await;
```
Comment (hiltontj) [inline influxdb3_write/src/write_buffer/mod.rs:1134]: 👍

**Review by hiltontj — APPROVED** [review]
Some minor suggestions otherwise LGTM

**Review by pauldix — APPROVED** [review]
With Trevor's small suggested changes, LGTM!

**Pull request merged**